3rdParty/velocypack/src/Slice.cpp

```diff
@@ -949,12 +950,13 @@ int Slice::compareString(char const* value, size_t length) const {
   return res;
 }
 
-int Slice::compareStringUnchecked(char const* value, size_t length) const noexcept {
+int Slice::compareStringUnchecked(StringRef const& value) const noexcept {
+  size_t const length = value.size();
   ValueLength keyLength;
   char const* k = getStringUnchecked(keyLength);
   size_t const compareLength =
       (std::min)(static_cast<size_t>(keyLength), length);
```

Comment:
Don‘t you need to use checkOverflow here as well ?

Reply:
Not here, as `compareStringUnchecked` is meant to perform less checks, and is also marked `noexcept`. Thus it is intentionally not checking everything.
However, `compareString` was missing the check, and I will add it there.